pkg/plugins/github/changelog.go

```diff
@@ -17,70 +18,66 @@ type Changelog struct {
 	Report      string
 }
 
-// Changelog returns a changelog description based on a release name
-func (g *Github) Changelog(name string) (string, error) {
-
-	/*
-			https://developer.github.com/v4/explorer/
-		# Query
-		query getLatestRelease($owner: String!, $repository: String!){
-			repository(owner: $owner, name: $repository){
-				release(tagName: 1){
-					description
-					publishedAt
-					url
-				}
-			}
+// changelogQuery defines a github v4 API query to retrieve the changelog of a given release
+/*
+https://developer.github.com/v4/explorer/
+# Query
+query getLatestRelease($owner: String!, $repository: String!){
+	repository(owner: $owner, name: $repository){
+		release(tagName: "v0.17.0"){
+			description
+			publishedAt
+			url
 		}
-		# Variables
-		{
-			"owner": "olblak",
-			"repository": "charts",
-		}
-	*/
+	}
+}
+# Variables
+{
+	"owner": "updatecli",
+	"repository": "updatecli"
+}
+*/
+type changelogQuery struct {
+	Repository struct {
+		Release queriedRelease `graphql:"release(tagName: $tagName)"`
+	} `graphql:"repository(owner: $owner, name: $repository)"`
+}
+type queriedRelease struct {
+	Description string
+	Url         string
+	PublishedAt time.Time
+}
 
-	client := g.NewClient()
+// Changelog returns a changelog description based on a release name
+func (g *Github) Changelog(version version.Version) (string, error) {
```

Comment:
Without being a blocker, for now, I think it wouldn't have been better to create a new struct directly in the changelog package such as "changelogInformation or "changelogVersion"?
We now have an unnecessary strong binding between this package and version.

Reply:
Not sure to understand?

- Using a private struct for a public method is not possible (how would the externe caller be able to use the struct)?
- The intent is to change the signature of the method `Changelog` globally (see the changes on the `Changelog` interface below) and use the version.Version as argument type. Rationale is that it depends on the implementation of the changelog if either the parsed or originl version are required.

Reply:
I was thinking to create a public struct defined in the changelog package so we wouldn't need to import the package version. But the way you implemented it is very good